tests/cache_control.rs

```diff
@@ -1,34 +1,95 @@
 use momento::requests::cache::create_cache::CreateCache;
+use momento::requests::cache::{
+    basic::{
+        get::{Get, GetValue},
+        set::Set,
+    },
+    flush_cache::FlushCache,
+};
+use momento::MomentoResult;
 use uuid::Uuid;
 
 use momento_test_util::CACHE_TEST_STATE;
 
 #[tokio::test]
-async fn delete_nonexistent_cache_returns_not_found() {
-    let client = CACHE_TEST_STATE.client.clone();
+async fn delete_nonexistent_cache_returns_not_found() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
     let cache_name = "fake-cache-".to_string() + &Uuid::new_v4().to_string();
     let result = client.delete_cache(cache_name).await.unwrap_err();
     let _err_msg = "not found".to_string();
-    assert!(matches!(result.to_string(), _err_message))
+    assert!(matches!(result.to_string(), _err_message));
+    Ok(())
 }
 
 #[tokio::test]
-async fn create_existing_cache_returns_already_exists() {
-    let client = CACHE_TEST_STATE.client.clone();
-    let cache_name = CACHE_TEST_STATE.cache_name.clone();
-    let result = client.create_cache(cache_name).await.unwrap();
+async fn create_existing_cache_returns_already_exists() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = &CACHE_TEST_STATE.cache_name;
+    let result = client.create_cache(cache_name).await?;
     assert_eq!(result, CreateCache::AlreadyExists {});
+    Ok(())
 }
 
 #[tokio::test]
-async fn lists_existing_test_cache() {
-    let client = CACHE_TEST_STATE.client.clone();
-    let cache_name = CACHE_TEST_STATE.cache_name.clone();
-    let result = client.list_caches().await.unwrap();
+async fn lists_existing_test_cache() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = &CACHE_TEST_STATE.cache_name;
+    let result = client.list_caches().await?;
     let cache_names: Vec<String> = result
         .caches
         .iter()
         .map(|cache_info| cache_info.name.clone())
         .collect();
-    assert!(cache_names.contains(&cache_name));
+    assert!(cache_names.contains(cache_name));
+    Ok(())
+}
+
+#[tokio::test]
+async fn flush_nonexistent_cache_returns_not_found() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = "fake-cache-".to_string() + &Uuid::new_v4().to_string();
+    let result = client.flush_cache(cache_name).await.unwrap_err();
+    let _err_msg = "not found".to_string();
+    assert!(matches!(result.to_string(), _err_message));
+    Ok(())
+}
+
+#[tokio::test]
+async fn flush_existing_cache_returns_success() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = &CACHE_TEST_STATE.cache_name;
+
+    // Insert some elements
+    let set_result1 = client.set(cache_name.to_string(), "key1", "value1").await?;
```

Comment:
I really want to figure out how to prevent users from needing to do this `to_string()` call here. This isn't an issue with your PR most likely, probably something with how we are defining the signature of the function. What error do you get if you don't do the `to_string()`?

Reply:
it's fine without `to_string()`, might've been leftover from when `cache_name` was cloned instead of borrowed or something. will remove those `to_string()` calls